extensions/eclipse-che-theia-workspace/src/browser/che-quick-open-workspace.ts

```diff
@@ -0,0 +1,229 @@
+
+    protected async open(workspaces: cheApi.workspace.Workspace[]): Promise<void> {
+        this.items = [];
+
+        if (!workspaces.length) {
+            this.items.push(new QuickOpenGroupItem({
+                label: 'No Recent Workspaces',
+                run: (mode: QuickOpenMode): boolean => false
+            }));
+            return;
+        }
+
+        for (const workspace of workspaces) {
+            const icon = this.labelProvider.folderIcon;
+            const iconClass = icon + ' file-icon';
+            this.items.push(new QuickOpenGroupItem({
+                label: this.getWorkspaceName(workspace) + (this.isCurrentWorkspace(workspace) ? ' (Current)' : ''),
+                description: this.getWorkspaceStack(workspace),
+                groupLabel: `last modified ${moment(this.getWorkspaceModificationTime(workspace)).fromNow()}`,
+                iconClass,
+                run: (mode: QuickOpenMode): boolean => {
+                    if (mode !== QuickOpenMode.OPEN) {
+                        return false;
+                    }
+
+                    if (this.isCurrentWorkspace(workspace)) {
+                        return true;
+                    }
+
+                    this.openWorkspace(workspace);
+
+                    return true;
+                },
+            }));
+        }
+
+        this.quickOpenService.open(this, {
+            placeholder: 'Type the name of the Che workspace you want to open',
+            fuzzyMatchLabel: true,
+            fuzzySort: false
+        });
+    }
+
+    onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): void {
+        acceptor(this.items);
+    }
+
+    async select(): Promise<void> {
+        this.items = [];
+
+        const token = await this.oAuthUtils.getUserToken();
+
+        if (!this.currentWorkspace) {
+            this.currentWorkspace = await this.cheApi.currentWorkspace();
+        }
+
+        if (!this.currentWorkspace.namespace) {
+            return;
+        }
+
+        const workspaces = await this.cheApi.getAllByNamespace(this.currentWorkspace.namespace, token);
+
+        workspaces.sort((a: cheApi.workspace.Workspace, b: cheApi.workspace.Workspace) => {
+            const updatedA: number = this.getWorkspaceModificationTime(a);
+            const updatedB: number = this.getWorkspaceModificationTime(b);
+
+            if (isNaN(updatedA) || isNaN(updatedB)) {
+                return 0;
+            } else {
+                return updatedB - updatedA;
+            }
+        });
+
+        await this.open(workspaces);
+    }
+
+    private getWorkspaceName(workspace: cheApi.workspace.Workspace): string | undefined {
+        if (workspace.config) {
+            return workspace.config.name;
+        } else if (workspace.devfile && workspace.devfile.metadata) {
+            return workspace.devfile.metadata.name;
+        }
+    }
+
+    private getWorkspaceStack(workspace: cheApi.workspace.Workspace): string | undefined {
```

Comment:
AFAIK, this can be reached when user creates a workspace from custom devfile, editing directly from the dashboard, not importing the last one from URL.